e2e/azure-resources/azure-container-registry-airgap.json

```diff
@@ -0,0 +1,48 @@
+{
```

Comment:
The json file gets auto-generated during createAzureResourceWithBicep

Reply:
could we consider just using an ARM template and getting rid of the bicep so we don't have to delegate out to `az bicep build` during every test run?

Reply:
we might also consider deploying all the other airgap-related resources via an ARM/bicep template as well

Reply:
Sylvain suggested using bicep so we should talk with him about ARM vs bicep. And I agree, I think a separate PR / task can be created for uniformly creating all the resources, depending on which direction we want to go in.

Reply:
I will say, I do think bicep files are a bit easier to create and read over the arm/json files. But it does come with the trade off of needing to run `az bicep build`

Reply:
we should indeed only keep the .bicep file, and run `az bicep build` and I didn't expect that resources were all managed in Go, so it does make it look weird that we have go code to generate half (if not all) of the resource and the ACR is in Bicep... We should definitely look at unifying.  I do have a stronger opinion on the Bicep logic since we benefits from others (Azure Verified Module - AVM team) to augment the module with better security and compliance and we will simply need to bump the versions over time, VS having to update the Go package, SDK, find the new parameters, etc (last time I needed to set auto-upgrade on the AKS node pool, I had to do a major go module bump from v1 to v6..)